faster_sam/middleware.py

```diff
@@ -5,10 +5,48 @@
 
 
 class RemovePathMiddleware(BaseHTTPMiddleware):
+    """
+    Middleware for removing a specified part of the request path.
+
+    e.g
+
+    >>> app = FastAPI()
+    >>> app.add_middleware(RemovePathMiddleware, path="/foo")  # Removes /foo from requests path
+    >>> @app.get("/bar")
+    ... def bar():
+    ...     return {"message": "Responding to GET /foo/bar}
```

Comment:
Aqui seria return {"message": "Responding to GET /bar}, não? pois remove o "/foo" antes de resolver a rota.

/foo/var ---> midleware ----> /bar

Reply:
@gabriel-f-santos o exemplo seria para ilustrar que apesar de estar registrada a rota apenas como `/bar` se chamado via `/foo/bar` também executaria esse endpoint. Acha que ficou confuso, consegue pensar em um exemplo melhor?

Reply:
Acho que adicionar essa etapa /foo/var ---> midleware ----> /bar fica melhor de entender, aí poderia manter a linha 17 do jeito que está

Reply:
@gabriel-f-santos @julianolf Seria algo assim certo?

`   >>> app = FastAPI()
    >>> app.add_middleware(RemovePathMiddleware, path="/foo/bar")  # Removes /foo from requests path
    >>> @app.get("/bar")
    ... def bar():
    ...     return {"message": "Responding to GET /bar}`

Reply:
>>> app = FastAPI()
>>> app.add_middleware(RemovePathMiddleware, path="/foo/bar")  # Removes /foo from requests path
>>>  /foo/var ---> midleware ----> /bar
>>> @app.get("/bar")
... def bar():
... return {"message": "Responding to GET /foo/bar}

Tinha pensado desse jeito, adicionando uma linha explicando o fluxo